### DISCUSSION TITLE
How about allowing snippets with imports or library dependencies eg by adding expansion complete hooks?

### DISCUSSION THREAD

**Discussion by MarcWeber** [dcampos/nvim-snippy#130] [Ideas]
when I asked at liberachat the reply was should be part of language server.
But the language server's job is not to know what I mean with "cow()" function.
Could be anything. Even worse: Just adding to package.json does not do it for me because I replaced it by my own code which generates that.

Now you told me you want to be compatible to standard so question arises where such could be discussed and how format might change ?

Some non standard extension like this would do it for me. Maybe it can be already done in a different way:

```
snippet X ""  action=lua:foo()
     code here
````
or similar. Maybe it can already be done and I missed it? I searched for hook, event. There is complete_done but no expansion_done(snippet) or such yet.

I know that expansion complete might not be well defined. No action could be any like action_snipped_expanded or action_snipped_complete or whatever makes sense. For my use case doesn't really matter.

Eventually language servers could be asked to 'add the imports' where they think makes sense cause existing import lines often can be reused.

**Comment by dcampos** [comment]
I'm not sure if it will be enough for your use case, but we have some `autocmds` available (see `h: snippy-autocmds` for more info). Maybe `SnippyFinished` could be useful.
